tools/loader/ImageClassifier.cpp

```diff
@@ -364,6 +372,76 @@ static void parseInputImageList(const std::string &inputImageListFile) {
   inFile.close();
 }
 
+/// Run inference request on HostManager.
+static void runInference(runtime::HostManager *hostManager, std::string name,
+                         std::unique_ptr<ExecutionContext> batch,
+                         std::promise<void> &runPromise,
+                         std::atomic<unsigned> &inflight,
+                         std::atomic<int> &dispatched) {
+  hostManager->runNetwork(name, std::move(batch),
+                          [&runPromise, &inflight, &dispatched, hostManager,
+                           name](runtime::RunIdentifierTy, llvm::Error err,
+                                 std::unique_ptr<ExecutionContext> contextPtr) {
+                            EXIT_ON_ERR(std::move(err));
+                            // Kick off another run.
+                            if (dispatched.fetch_sub(1) > 0) {
+                              inflight++;
+                              runInference(hostManager, name,
+                                           std::move(contextPtr), runPromise,
+                                           inflight, dispatched);
+                            }
+                            if (--inflight == 0) {
+                              runPromise.set_value();
+                            }
+                          });
+}
+
+/// Run benchmark and wait for all runs to complete.
+static void
+runBenchmark(std::string name, Loader &loader,
+             std::vector<std::unique_ptr<ExecutionContext>> contexts,
+             unsigned requestCount) {
+  runtime::HostManager *hostManager = loader.getHostManager();
+  std::atomic<unsigned> inflight(0);
+  std::atomic<int> dispatched(requestCount);
+  std::promise<void> runPromise;
+  auto fut = runPromise.get_future();
+
+  // Kick off initial pool of requests.
+  for (unsigned i = 0; i < contexts.size(); i++) {
```

Comment:
nit: `size_t` instead of `unsigned`. Also, follow for loop pattern from other places, e.g. `size_t i = 0, e = contexts.size(); i < e; i++`